src/process.ts

```diff
@@ -48,19 +48,37 @@ export const handleProcessCommand = async (
     };
   }
   if ("approved" in referendum && referendum.approved) {
-    await requestState.octokitInstance.rest.pulls.merge({
+    try {
+      await requestState.octokitInstance.rest.pulls.merge({
+        owner: githubActions.context.repo.owner,
+        repo: githubActions.context.repo.repo,
+        pull_number: githubActions.context.issue.number,
+        merge_method: "squash",
+      });
+      return { success: true, message: "The on-chain referendum has approved the RFC." };
+    } catch (e) {
+      requestState.octokitInstance.log.error(String(e));
+      return {
+        success: false,
+        errorMessage:
+          "The on-chain referendum has approved the RFC, however I was not able to merge the PR automatically.",
+      };
+    }
+  }
+  try {
+    await requestState.octokitInstance.rest.pulls.update({
       owner: githubActions.context.repo.owner,
       repo: githubActions.context.repo.repo,
       pull_number: githubActions.context.issue.number,
-      merge_method: "squash",
+      state: "closed",
```

Comment:
Does it close the Pull request? Is it desired to close, or might they have additional work and another round of vote?

Reply:
Yes, we close it because in this situation it's not just a missing approval, in this situation we have an explicit `RFC_REJECT` referendum which passed. I pushed a commit trying to make it a little more explicit.

Reply:
Here is some more explanation:

https://github.com/paritytech/rfc-action/blob/708c628d1cd9aaada855f289e49218b3b644ad6c/src/find-referendum-state.ts#L9-L15